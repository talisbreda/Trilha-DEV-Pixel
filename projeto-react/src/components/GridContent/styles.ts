import styled, { css } from 'styled-components';
import { Theme } from '../../styles/theme';

interface Props {
  theme: Theme;
}

export const Container = styled.div<Props>`
  ${({ theme }) => css`
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
  `}
`;

export const Html = styled.div<Props>`
  ${({ theme }) => css`
    margin: ${theme.spacings.huge} 0;
  `}
`;
