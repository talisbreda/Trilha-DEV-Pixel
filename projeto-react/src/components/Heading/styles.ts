import styled, { css, FlattenSimpleInterpolation } from 'styled-components';
import { Theme } from '../../styles/theme';

interface Props {
  theme: Theme;
  darkColor: boolean;
  as: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
  size: 'huge' | 'large' | 'medium' | 'small';
  upperCase: boolean;
}

type sizes = {
  small: (theme: Theme) => FlattenSimpleInterpolation;
  medium: (theme: Theme) => FlattenSimpleInterpolation;
  large: (theme: Theme) => FlattenSimpleInterpolation;
  huge: (theme: Theme) => FlattenSimpleInterpolation;
};

const mediaFont = (theme: Theme) => css`
  @media ${theme.media.ltMedium} {
    font-size: ${theme.font.sizes.large}
  }
`;

const titleSize: sizes = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.small};
  `,
  medium: (theme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  large: (theme) => css`
    font-size: ${theme.font.sizes.large};
  `,
  huge: (theme) => css`
    font-size: ${theme.font.sizes.huge};
    ${mediaFont(theme)};
  `,
};

const titleCase = (upperCase: boolean) => css`
  text-transform: ${upperCase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1<Props>`
  ${({ theme, darkColor, size, upperCase }) => {
    return css`
        color: ${darkColor ? theme.colors.primaryColor : theme.colors.white};
        ${titleSize[size](theme)};
        ${titleCase(upperCase)};
    `;
  }}
`;
