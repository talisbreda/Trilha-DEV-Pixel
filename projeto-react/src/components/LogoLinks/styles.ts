import styled, { css } from 'styled-components';
import { Theme } from '../../styles/theme';

interface Props {
  theme: Theme;
  href: string;
}

export const Container = styled.a<Props>`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    text-decoration: none;
    color: inherit;

    > img {
      max-height: 3rem;
    }
  `}
`;
