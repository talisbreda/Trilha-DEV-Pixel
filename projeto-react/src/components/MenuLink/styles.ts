import styled, { css } from 'styled-components';
import { Theme } from '../../styles/theme';

interface Props {
  theme: Theme;
  href: string;
  target: string;
}

export const Container = styled.a<Props>`
  ${({ theme }) => css`
    display: block;
    text-decoration: none;
    font-size; ${theme.font.sizes.tiny};
    padding: ${theme.spacings.small};
    color: ${theme.colors.primaryColor};
    position: relative;

    &::after {
      content: '';
      position: absolute;
      bottom: 0.75rem;
      left: 50%;
      width: 0;
      height: 0.2rem;
      background: ${theme.colors.secondaryColor};
      transition: all 300ms ease-in-out;
    }
    &:hover::after {
      left: 25%;
      width: 50%;
    }
  `}
`;
