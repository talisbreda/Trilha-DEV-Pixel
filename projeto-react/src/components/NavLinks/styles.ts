import styled, { css } from 'styled-components';
import { Theme } from '../../styles/theme';

interface Props {
  theme: Theme;
}

export const Container = styled.nav<Props>`
  ${({ theme }) => css`
    background: #fff;
    display: flex;
    flex-flow: row wrap;

    @media ${theme.media.ltMedium} {
      flex-flow: column nowrap;
      align-items: center;
    }
  `}
`;
