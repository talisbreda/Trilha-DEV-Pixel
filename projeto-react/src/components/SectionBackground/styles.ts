import styled, { css } from 'styled-components';
import { Theme } from '../../styles/theme';

interface Props {
  theme: Theme;
  background: boolean;
}

const containerBackgroundActivate = (theme: Theme) => css`
  background: ${theme.colors.primaryColor};
  color: ${theme.colors.white};
`;

export const Container = styled.div<Props>`
  ${({ theme, background }) => css`
    background: ${theme.colors.white};
    color: ${theme.colors.primaryColor};
    ${background && containerBackgroundActivate(theme)};
    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;
