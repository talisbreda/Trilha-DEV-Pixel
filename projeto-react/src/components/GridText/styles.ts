import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';
import { Theme } from '../../styles/theme';
import { Title as HeadingContainer } from '../Heading/styles';

interface Props {
  theme: Theme;
}

export const Container = styled.div<Props>`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div<Props>`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large}
  `}
`;

export const GridElement = styled.div<Props>`
  ${({ theme }) => css`
    ${HeadingContainer} {
      position: relative;
      left: 5rem;
    }

    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: 7rem;
      top: -3rem;
      left: -5rem;
      transform: rotate(5deg);
    }
  `}
`;
