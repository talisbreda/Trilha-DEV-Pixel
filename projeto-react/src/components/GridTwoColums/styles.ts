import styled, { css } from 'styled-components';
import { Theme } from '../../styles/theme';
import { Title } from '../Heading/styles';

interface Props {
  theme: Theme;
}

export const Container = styled.div<Props>`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacings.large};

    @media ${theme.media.ltMedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }
    ${Title} {
      margin-bottom: ${theme.spacings.huge};
    }
  `}
`;

export const TextContainer = styled.div<Props>`
  ${({ theme }) => css`
    @media ${theme.media.ltMedium} {
      margin-bottom: ${theme.spacings.large};
    }
  `}
`;

export const ImageContainer = styled.div<Props>`
  ${() => css``}
`;

export const Image = styled.img<Props>`
  ${() => css``}
`;
