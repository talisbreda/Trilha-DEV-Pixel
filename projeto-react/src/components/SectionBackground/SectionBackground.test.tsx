import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { SectionBackground } from '.';

describe('<SectionBackground />', () => {
  it('should render with dark background', () => {
    const { container } = renderTheme(
      <SectionBackground background>
        <h1>Children</h1>
      </SectionBackground>,
    );
    expect(screen.getByRole('heading')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render with light background', () => {
    renderTheme(
      <SectionBackground background={false}>
        <h1>Children</h1>
      </SectionBackground>,
    );
    expect(screen.getByRole('heading')).toBeInTheDocument();
  });
});
