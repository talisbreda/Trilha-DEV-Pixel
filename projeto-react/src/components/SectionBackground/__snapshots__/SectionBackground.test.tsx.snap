// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionBackground /> should render with dark background 1`] = `
<div>
  <div
    class="sc-gswNZR iCCzcB"
  >
    <div
      class="sc-bcXHqe ilsefL"
    >
      <h1>
        Children
      </h1>
    </div>
  </div>
</div>
`;
